pkg/controller/infinispan/resources/secret/secret_controller.go

```diff
@@ -124,6 +141,99 @@ func (s secretResource) createSecret(name, label string, identities []byte) erro
 	return nil
 }
 
+func (s secretResource) reconcileKeystoreSecret() (*reconcile.Result, error) {
+	i := s.infinispan
+	if !i.IsEncryptionEnabled() {
+		return nil, nil
+	}
+
+	keystoreSecret := &corev1.Secret{}
+	if result, err := kube.LookupResource(i.GetKeystoreSecretName(), i.Namespace, keystoreSecret, s.client, s.log); result != nil {
+		return result, err
+	}
+
+	_, err := controllerutil.CreateOrUpdate(context.TODO(), s.client, keystoreSecret, func() error {
+		if ownerRef := metav1.GetControllerOf(keystoreSecret); ownerRef != nil && ownerRef.UID != s.infinispan.UID {
+			return fmt.Errorf("Unable to SetControllerReference for secret %s as it's already owned  by %s:%s:%s",
+				keystoreSecret.Name, ownerRef.Kind, ownerRef.Name, ownerRef.UID)
+		}
+		return controllerutil.SetControllerReference(i, keystoreSecret, s.scheme)
+	})
+	if err != nil {
+		return &reconcile.Result{}, err
+	}
+	return nil, nil
+}
+
+func (s secretResource) reconcileTruststoreSecret() (*reconcile.Result, error) {
+	i := s.infinispan
+	if !i.IsClientCertEnabled() {
+		return nil, nil
+	}
+
+	trustSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      i.GetTruststoreSecretName(),
+			Namespace: i.Namespace,
+		},
+	}
+
+	if trustSecret.Name == "" {
```

Comment:
Seems to `i.IsClientCertEnabled()` means that `i.GetTruststoreSecretName()` is not empty. Or I'm wrong?

Reply:
`IsClientCertEnabled()` only checks to see if the ClientCert type != `None`, it doesn't take into account whether the user has specified `ClientCertSecretName`. I could make it so that `IsClientCertEnabled` also checks the `ClientCertSecretName` value, but if I do that the user receives no feedback if they just set `ClientCert` and client cert will not be enabled as `ClientCertSecretName` is missing.